src/udev/udev-node.c

```diff
@@ -175,63 +175,43 @@ static int stack_directory_read_one(int dirfd, const char *id, bool is_symlink,
         return 1; /* Updated */
 }
 
-static int stack_directory_find_prioritized_devnode(sd_device *dev, const char *dirname, bool add, char **ret) {
+static int stack_directory_find_prioritized_devnode(int dirfd, char **ret) {
         _cleanup_closedir_ DIR *dir = NULL;
         _cleanup_free_ char *devnode = NULL;
-        int r, priority = 0;
-        const char *id;
+        char *candidate = NULL;
+        int r;
 
-        assert(dev);
-        assert(dirname);
+        assert(dirfd >= 0);
         assert(ret);
 
-        /* Find device node of device with highest priority. This returns 1 if a device found, 0 if no
-         * device found, or a negative errno on error. */
+        /* Find device node of device with the highest priority. If a candidate is found it's returned via
+         * 'ret' otherwise if no more device left 'ret' is set to NULL. On error this function returns a
+         * negative errno. */
 
-        if (add) {
-                const char *n;
-
-                r = device_get_devlink_priority(dev, &priority);
-                if (r < 0)
-                        return r;
-
-                r = sd_device_get_devname(dev, &n);
-                if (r < 0)
-                        return r;
-
-                devnode = strdup(n);
-                if (!devnode)
-                        return -ENOMEM;
-        }
-
-        dir = opendir(dirname);
+        dir = opendir(FORMAT_PROC_FD_PATH(dirfd));
         if (!dir)
                 return -errno;
 
-        r = device_get_device_id(dev, &id);
-        if (r < 0)
-                return r;
-
         FOREACH_DIRENT_ALL(de, dir, break) {
-                if (de->d_name[0] == '.')
-                        continue;
+                int priority;
```

Comment:
No no, `stack_directory_read_one()` _updates_ `devnode` and `priority`, if the entry has higher priority. Hence, `priority` must be declared and initialized before the loop.

Reply:
ouch... fixed